internal/controllers/synthesis/status.go

```diff
@@ -42,28 +43,22 @@ func (c *statusController) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, nil
 	}
 	if pod.Annotations == nil {
-		return ctrl.Result{}, nil
+		logger.V(1).Info("synthesizer pod without any annotations was found - removing its finalizer")
+		return c.removeFinalizer(ctx, pod)
 	}
 
 	comp := &apiv1.Composition{}
 	comp.Name = pod.OwnerReferences[0].Name
 	comp.Namespace = pod.Namespace
 	err = c.client.Get(ctx, client.ObjectKeyFromObject(comp), comp)
-	if err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting composition resource: %w", err))
-	}
-	if comp.Spec.Synthesizer.Name == "" {
-		return ctrl.Result{}, nil
+	if errors.IsNotFound(err) {
+		logger.V(1).Info("composition was deleted unexpectedly - releasing synthesizer pod")
+		return c.removeFinalizer(ctx, pod)
```

Comment:
The description says "missing synthesizers" does deleting a synth clean up all the compositions that depend on it?

Reply:
It doesn't because that seems... dangerous. Although it wouldn't do any harm to hold a finalizer until the synthesizer is no longer referenced by any compositions. Thoughts?

Reply:
Yup sound like a footgun, but I like the idea of holding a finalizer on the synth...